geopandas/tests/test_geodataframe.py

```diff
@@ -1220,6 +1220,25 @@ def test_repeat_geo_col(self):
         with pytest.raises(ValueError):
             GeoDataFrame(df3, geometry="geom")
 
+    @pytest.mark.parametrize("dtype", ["geometry", "object"])
+    def test_multiindex_with_geometry_label(self, dtype):
+        # DataFrame with MultiIndex where "geometry" label corresponds to
+        # multiple columns
+        df = pd.DataFrame([[Point(0, 0), Point(1, 1)], [Point(2, 2), Point(3, 3)]])
+        df = df.astype(dtype)
+        df.columns = pd.MultiIndex.from_product([["geometry"], [0, 1]])
+        # don't error in constructor
+        gdf = GeoDataFrame(df)
+        # Getting the .geometry column gives GeoDataFrame for both columns
+        # (but with first MultiIndex level removed)
+        # TODO should this give an error instead?
+        result = gdf.geometry
```

Comment:
The current behaviour is because of

```
    def _get_geometry(self):
        if self._geometry_column_name not in self:
            ... (raise errors)
        return self[self._geometry_column_name]
```

Because `self._geometry_column_name` is "geometry" (and thus found as present), we basically return `gdf["geometry"]`, which in pandas will select out the first MultiIndex level.

This is probably a broader question than this regression fix, since it was already like this before as well (so for a separate issue), but triggered by writing this test, I am wondering:

- Should we maybe (long term) guarantee that `.geometry` always returns a GeoSeries? (so in case it would return a GeoDataFrame, raise an error instead. Similarly as we started raising an error for duplicate "geometry" columns)
- Should the default `_geometry_column_name` actually still be set to "geometry" if that column is not present as a single geometry-dtype column? Maybe we should rather set it to None?

Reply:
Thanks for the PR, I was also having a quick look into this. 
>Should we maybe (long term) guarantee that .geometry always returns a GeoSeries?

I think this makes sense - the natural expectation is that calling .geometry will return something that calling a geometry method like `area`, `x/y` would apply to. We just have to make sure that doing this doesn't break multiindex generated internally by pandas (from e.g. unstack, pivot). Agree that this should be a separate issue / PR (it's generally tied into the less than ideal handling of MultiIndex columns).
>Should the default _geometry_column_name actually still be set to "geometry" if that column is not present as a single geometry-dtype column? Maybe we should rather set it to None?

This sounds reasonable, except perhaps for the case of #2088 where multiindex data is passed to the gdf constructor and the geometry data is added by the `geometry` keyword.